dask/array/blockwise.py

```diff
@@ -268,7 +269,10 @@ def blockwise(
             if ind in adjust_chunks:
                 if callable(adjust_chunks[ind]):
                     chunks[i] = tuple(map(adjust_chunks[ind], chunks[i]))
-                elif isinstance(adjust_chunks[ind], numbers.Integral):
+                elif isinstance(adjust_chunks[ind], numbers.Integral) or (
+                    isinstance(adjust_chunks[ind], numbers.Real)
```

Comment:
What is the numbers.Real doing/catching? I think we are normally just checking if the element is nan.

Reply:
If you don't require `numbers.Real` and then check for Nan then this may try to `isnan` a tuple and throw an error! I added this change because it was otherwise failing tests.

Reply:
Oh, I meant np.isscalar(...) and isnan(...)

Reply:
Ah - got you. Though `nan` should technically not be a scalar, I guess? 

I agree this is a bit more clear in general.

Reply:
nan is considered a scalar by numpy, that's how we typically check for it is a simple isnan check isn't enough

Reply:
Sure, the behavior of `np.isscalar` is manifest, was just philosophizing. I can also see where something that's a float not being a scalar in certain cases just making life overly problematic.

I think we should be good with this latest commit, aside from flaky tests.